grails-app/services/org/pih/warehouse/inventory/ProductAvailabilityService.groovy

```diff
@@ -1025,6 +1025,21 @@ class ProductAvailabilityService {
         return quantityAvailableToPromiseByProductNotInBin ?: 0
     }
 
+    Integer getQuantityAvailableToPromiseForProductInBin(InventoryItem inventoryItem, Location location, Location binLocation) {
+        return ProductAvailability.createCriteria().get {
+            projections {
+                property("quantityAvailableToPromise")
```

Comment:
@jmiranda this is the issue I've once fixed: https://pihemr.atlassian.net/browse/OBPIH-6008
I've run your query on my local db, and then checked all the columns how they look like.
Here is an example:
```
id                                  |version|bin_location_id                 |inventory_item_id               |location_id                     |product_id|product_code|lot_number  |bin_location_name|quantity_allocated|quantity_on_hand|date_created       |last_updated       |quantity_on_hold|quantity_available_to_promise|
------------------------------------+-------+--------------------------------+--------------------------------+--------------------------------+----------+------------+------------+-----------------+------------------+----------------+-------------------+-------------------+----------------+-----------------------------+
0ac391ad-4d6d-4a27-bc01-77139d1fd332|      0|09f5e116831aefb00183377b1e5d58c4|09f5e1167c3c9801017c4ba7755715fe|09f5e1167e6ee8ec017e9c3c480f6f43|10001     |21064       |MWM062620N47|R-307CFC         |                 0|               0|2023-08-30 18:09:01|2023-08-30 18:09:01|               0|                            0|
f47db6bf-6ee9-4d73-8a3a-f7c58e39f5dc|     22|09f5e116831aefb00183377b1e5d58c4|09f5e1167c3c9801017c4ba7755715fe|09f5e1167e6ee8ec017e9c3c480f6f43|10001     |10001       |MWM062620N47|R-307CFC         |                 0|               0|2023-09-18 21:01:21|2023-10-25 21:18:03|               0|                            0|
```
so what needs to be done on production is to refresh PA on the locations where those rows exist

Reply:
@kchelstowski Although you've fixed the data issue, the problem is that this scenario can still happen. So we need to understand how it happens and be defensive about how to handle it.

@awalkowiak Let me know if you have any thoughts. It might not be fixable but we shouldn't ignore it if it's a possibility.